wrangler-service/src/main/java/co/cask/wrangler/service/spanner/SpannerSpecification.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.wrangler.service.spanner;
+
+import co.cask.cdap.api.data.schema.Schema;
+
+/**
+ * Spanner specification properties for spanner source plugin
+ */
+public class SpannerSpecification {
+  // gcp properties
+  private final String serviceFilePath;
+  private final String project;
+  // spanner properties
+  private final String instance;
+  private final String database;
+  private final String table;
+  private final Schema schema;
```

Comment:
I assume this means the other ones use the actual Schema class instead of the string version? It's really weird, since the plugin actually expects a string, which means the UI is doing some conversion?

Reply:
Other classes, construct the json object manually, however they convert schema to string and send it, i can store it as string too.